compiler/src/iree/compiler/Codegen/Dialect/IREECodegenAttrs.cpp

```diff
@@ -126,6 +86,71 @@ LogicalResult TranslationInfoAttr::verify(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// iree_codegen.lowering_config_level
+//===----------------------------------------------------------------------===//
+
+void LoweringConfigTilingLevelAttr::print(mlir::AsmPrinter &printer) const {
+  auto tileInterchange = getInterchange();
```

Comment:
I am curious why the assembly format parser/printer didnt work here?

Reply:
Ideally you could write:
```
let assemblyFormat = [{
    (`{` `sizes` `=` `[` $sizes `]` `,` `interchange` `=` `[` $interchange^ `]` `}`)
    : (`[` $sizes `]`)?
}];
```
But unfortunately the `assemblyFormat` limits parameters to appearing once (even across groups), so this results in: `error: duplicate parameter 'sizes'`.

Reply:
I cant parse the `assemblyFormat` above, but are you trying to print/parse the `sizes` and `interchange` for each level at the same time?

Reply:
With the new structure, the tiling levels are a list of `TilingLevelAttr`s, which contains both the sizes and interchange for a particular level. This makes the sizes and interchange 1D parameters to the `TilingLevelAttr`, which means they can be typed arrays (they have to be 1D as the typed arrays only support primitive values). 

The custom format is to make updating the tests easier (though I think looks pretty nice too :))

You could use the generated printer/parser here, but that'd only allow the format to be:
```
 #iree_codegen.lowering_config<tiling_levels = [
   {sizes=[32, 64, 0]}, {sizes=[8, 32, 0]}, {sizes=[0, 0, 16]}]>
```
The custom printer allows special-casing the "no tile interchange" case as:
```
 // Pretty much the same as the old syntax:
 #iree_codegen.lowering_config<tiling_levels = [
   [32, 64, 0], [8, 32, 0], [0, 0, 16]]>
```
This makes most tests much easier to update (just `s/tile_sizes/tiling_levels`). The full `{sizes=[32, 64, 0], interchange=[1, 0, 2]}` is only used in a handful of tests as most places don't use `tile_interchange`.  

Hope that makes sense!

Reply:
Thanks for the details! THat makes sense.